src/walletinitinterface.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef WALLETINITINTERFACE_H
+#define WALLETINITINTERFACE_H
+
+#include <string>
+
+class CScheduler;
+class CRPCTable;
+
+class WalletInitInterface {
```

Comment:
Every method name in this interface has "Wallet" or "Wallets" in it, which is rather redundant given that this is a **Wallet**InitInterface. I'd say drop them all, so "GetWalletHelpString" => "GetHelpString", "WalletParameterInteraction" => "CheckParameterInteraction", "RegisterWalletRPC" => "RegisterRPC".

That alone would go a long way towards making this a generalized init module interface.

Reply:
I've updated the function names, since as you point out, "Wallet" is redundant.

See my comment below about a generalized module init interface.